java/java.source.base/src/org/netbeans/api/java/source/ElementHandle.java

```diff
@@ -634,7 +634,11 @@ public ElementHandle create(ElementKind kind, String... descriptors) {
                     }
                     return new ElementHandle<VariableElement> (kind, descriptors);
                 default:
-                    throw new IllegalArgumentException ();
+                    if(kind.name().equals(TreeShims.RECORD) &&  (descriptors.length == 1)) {
+                        return new ElementHandle<TypeElement> (kind, descriptors);
+                    }
+                    else throw new IllegalArgumentException ();
```

Comment:
Please use `{}` (for `else`) and reformat your changes.

Reply:
comment addressed